src/pages/ServicePage.js

```diff
@@ -0,0 +1,58 @@
+/* @flow */
+
+import mui from "material-ui";
+import React from 'react';
+import Router from 'react-router';
+
+import iss from '../iss';
+
+class CategoryPage extends React.Component {
+    constructor(props: Object) {
+        super(props);
+        this.state = {
+            object: {
+                site: {},
+            },
+        };
+    }
+
+    /**
+     * search:
+     * Do a search in ISS
+     *
+     * Returns: a promise to an object of data from ISS.
+     */
+
+    componentDidMount(): void {
+        iss(`service/${this.props.params.id}/`)
```

Comment:
I'd prefer to create a 'loading' component which shows errors consistently, etc - then make all the other components synchronous.